cms/djangoapps/contentstore/tests/test_transcripts_utils.py

```diff
@@ -643,3 +644,133 @@ def test_unicode(self):
         self.assertEqual(name, u'subs_˙∆©ƒƒƒ.srt.sjson')
         name = transcripts_utils.subs_filename(u"˙∆©ƒƒƒ", 'uk')
         self.assertEqual(name, u'uk_subs_˙∆©ƒƒƒ.srt.sjson')
+
+
+@ddt.ddt
+class TestGetTranscriptsData(unittest.TestCase):
+    """
+    Tests for transcript utils
+    """
+    @ddt.data(
+        (
+            {
+                'edx_video_id': '000-000-000',
+                'youtube_id_1_0': '12as34',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm', 'foo/bar/baz.m3u8'
+                ],
+            },
+            lambda **kwargs: {
+                ('000-000-000', 'en'): 'EDX_VIDEO_ID_SUBS',
+                ('12as34', 'en'): 'YOUTUBE_SUBS',
+                ('foo', 'en'): 'HTML5_SUBS'
+            }[(kwargs['video_id'], kwargs['language_code'])],
+            'EDX_VIDEO_ID_SUBS',
+        ),
+        (
+            {
+                'edx_video_id': '',
+                'youtube_id_1_0': '12as34',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm', 'foo/bar/baz.m3u8'
+                ],
+            },
+            lambda **kwargs: {
+                ('12as34', 'en'): 'YOUTUBE_SUBS',
+                ('foo', 'en'): 'HTML5_SUBS'
+            }[(kwargs['video_id'], kwargs['language_code'])],
+            'YOUTUBE_SUBS',
+        ),
+        (
+            {
+                'edx_video_id': '',
+                'youtube_id_1_0': '',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm',
+                ],
+            },
+            lambda **kwargs: {
+                ('foo', 'en'): 'HTML5_SUBS'
+            }[(kwargs['video_id'], kwargs['language_code'])],
+            'HTML5_SUBS',
+        )
+    )
```

Comment:
@Qubad786 Any good reason for using this dark magic(`lambda **kwargs: {...}[(kwargs['video_id'], kwargs['language_code'])]`) above in  all `ddt` scenarios? Can not we just use `mock_is_transcript_available.return_value`? I would highly recommend to change this.

Reply:
Opss I don't see any black magic here 😄.  This this is just to make sure the return value at some arguments. This way we know exactly about what were the parameters and what was the return value when this util was called and when we know them we can pretty much predict about the expected value. Looks more realistic then a blind return. Also, in our case, It'll not make any sense to just return a value as it will be returned as is, no matter the input.